src/parser.rs

```diff
@@ -1741,12 +1783,77 @@ impl<'a> Parser<'a> {
                 if statement == until {
                     break
                 } else {
-                    statements.push(statement)
+                    match statement.clone() {
+                        Statement::Catch(code) => {
+                            match self.push_statement_or_return_fixed_code(
+                                statements,
+                                statement,
+                                code.clone(),
+                            ) {
+                                Some(fixed_code) => {
+                                    statements.push(Statement::Catch(fixed_code));
+                                }
+                                None => {}
+                            }
+                        }
+                        Statement::Else(code) => {
+                            match self.push_statement_or_return_fixed_code(
+                                statements,
+                                statement,
+                                code.clone(),
+                            ) {
+                                Some(fixed_code) => {
+                                    statements.push(Statement::Else(fixed_code));
+                                }
+                                None => {}
+                            }
+                        }
+                        Statement::ElseIf(condition, code) => {
+                            match self.push_statement_or_return_fixed_code(
+                                statements,
+                                statement,
+                                code.clone(),
+                            ) {
+                                Some(fixed_code) => {
+                                    statements.push(Statement::ElseIf(condition, fixed_code));
+                                }
+                                None => {}
+                            }
+                        }
+                        _ => {
+                            statements.push(statement);
+                        }
+                    }
                 }
             }
         }
     }
 
+    fn push_statement_or_return_fixed_code(
+        &self,
+        statements: &mut Vec<Statement>,
+        statement: Statement,
+        code: Vec<Statement>,
+    ) -> Option<Vec<Statement>> {
+        let mut comments = self.find_previous_comments(statements);
+        if comments.is_empty() {
```

Comment:
just return the vec whether its empty or not

Reply:
I think code will be faster in most cases with this condition